### DISCUSSION TITLE
Desk structure in sanity studio v3

### DISCUSSION THREAD

**Discussion by OtterArkar** [sanity-io/sanity#3932] [Sanity Studio Support] [answered]
How can i use deskStructure builder in sanity studio v3. I tried adding structure in deskTool options but not working.
![2022-12-02_13-55 (1)](https://user-images.githubusercontent.com/115974288/205676456-b1a24ac9-034b-4aac-8f54-e619aef535c4.png)

**Comment by joaisa17** [comment, marked as answer]
The structure resolver is a callback function that takes the `StructureBuilder` as its argument.

The following example is the base document type list:
```ts
import { defineConfig } from 'sanity';

defineConfig({
    name: 'sanity',
    title: 'Sanity Example',

    projectId: 'foo',
    dataset: 'production',

    plugins: [
        deskTool({
            structure: S => S.documentTypeListItems(),

            name: 'desk',
            title: 'Desk'
        })
    ]
});
```

**Reply by OtterArkar** [reply]
Thanks that works
